caja version 1.16 build fail with mate-desktop-1.16.0 in gtk+2

CC       check-program.o
  CCLD     libeel-2.la
  CCLD     check-program
./.libs/libeel-2.a(eel-background.o): In function `init_fade':
/root/mate2/mate-desktop-1.16.0/caja/eel/eel-background.c:585: undefined reference to `mate_bg_crossfade_set_start_surface'
/root/mate2/mate-desktop-1.16.0/caja/eel/eel-background.c:579: undefined reference to `mate_bg_get_surface_from_root'
/root/mate2/mate-desktop-1.16.0/caja/eel/eel-background.c:580: undefined reference to `mate_bg_crossfade_set_start_surface'
./.libs/libeel-2.a(eel-background.o): In function `eel_background_ensure_realized':
/root/mate2/mate-desktop-1.16.0/caja/eel/eel-background.c:440: undefined reference to `mate_bg_create_surface'
./.libs/libeel-2.a(eel-background.o): In function `set_root_surface':
/root/mate2/mate-desktop-1.16.0/caja/eel/eel-background.c:536: undefined reference to `mate_bg_set_surface_as_root'
/root/mate2/mate-desktop-1.16.0/caja/eel/eel-background.c:531: undefined reference to `mate_bg_create_surface'
./.libs/libeel-2.a(eel-background.o): In function `fade_to_surface':
/root/mate2/mate-desktop-1.16.0/caja/eel/eel-background.c:606: undefined reference to `mate_bg_crossfade_set_end_surface'
collect2: error: ld returned 1 exit status



gtk+2 version   2.24


<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/38291067-caja-version-1-16-build-fail-with-mate-desktop-1-16-0-in-gtk-2?utm_campaign=plugin&utm_content=tracker%2F651521&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F651521&utm_medium=issues&utm_source=github).
</bountysource-plugin>

--- a/configure.ac
+++ b/configure.ac
@@ -240,10 +240,12 @@ AC_ARG_ENABLE(libunique,
 msg_libunique=no
 if test "x$enable_libunique" != "xno"; then
     msg_libunique=yes
-    AM_CONDITIONAL([ENABLE_LIBUNIQUE], = "1")
+    AM_CONDITIONAL([ENABLE_LIBUNIQUE], true)
+    AC_DEFINE(ENABLE_LIBUNIQUE, 1, [define to enable libunique])
 else
     msg_libunique=no
-    AM_CONDITIONAL([ENABLE_LIBUNIQUE], = "0")
+    AM_CONDITIONAL([ENABLE_LIBUNIQUE], false)
+    AC_DEFINE(ENABLE_LIBUNIQUE, 0, [define to disable libunique])
 fi
 
 dnl =====

--- a/src/caja-application.h
+++ b/src/caja-application.h
@@ -27,6 +27,7 @@
 #ifndef CAJA_APPLICATION_H
 #define CAJA_APPLICATION_H
 
+#include <config.h>
 #include <gdk/gdk.h>
 #include <gio/gio.h>
 #include <gtk/gtk.h>

